src/hooks/use-download-json.ts

```diff
@@ -0,0 +1,25 @@
+import { useCurrentSnippetId } from "@/hooks/use-current-snippet-id"
+import { useSnippet } from "@/hooks/use-snippet"
+import { useMemo, useState } from "react"
+import { decodeUrlHashToText } from "@/lib/decodeUrlHashToText"
+import { useRunTsx } from "@/hooks/use-run-tsx"
+
+export const useDownloadJSON = () => {
+  const snippetId = useCurrentSnippetId()
+  const { data: snippet, isLoading } = useSnippet(snippetId)
+
+  const defaultCode = useMemo(() => {
+    return decodeUrlHashToText(window.location.toString()) ?? snippet?.code
+  }, [])
+
+  const [code, setCode] = useState(defaultCode ?? "")
+
+  const { message, circuitJson } = useRunTsx(code, snippet?.snippet_type)
```

Comment:
Very expensive recompute

Reply:
I've installed the pacakage `file-saver` which will install the files

Reply:
the issue here is your running the tsx every time the page renders just to display a download button

Reply:
i think the easiest solution is to have the `circuitJson` passed into the `DownloadButtonAndMenu` and remove this hook

Reply:
Cool! Thanks for the help let's do it this way.